streams/src/main/java/org/apache/kafka/streams/kstream/internals/ProcessDetails.java

```diff
@@ -0,0 +1,341 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals;
+
+
+import org.apache.kafka.common.serialization.Serde;
+import org.apache.kafka.common.serialization.Serializer;
+import org.apache.kafka.streams.Consumed;
+import org.apache.kafka.streams.Topology;
+import org.apache.kafka.streams.errors.TopologyException;
+import org.apache.kafka.streams.kstream.Materialized;
+import org.apache.kafka.streams.kstream.Produced;
+import org.apache.kafka.streams.processor.ProcessorSupplier;
+import org.apache.kafka.streams.processor.StateStore;
+import org.apache.kafka.streams.processor.StateStoreSupplier;
+import org.apache.kafka.streams.processor.StreamPartitioner;
+import org.apache.kafka.streams.processor.TimestampExtractor;
+import org.apache.kafka.streams.state.KeyValueStore;
+import org.apache.kafka.streams.state.StoreBuilder;
+
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+/**
+ * Class used to hold the implementation details when making calls to inheritors of
+ * {@link AbstractStream} when building a topology
+ */
+@SuppressWarnings({"deprecation", "unchecked"})
+class ProcessDetails<K, V, S extends StateStore> {
+
+    private final Collection<String> sourceTopics;
+    private final String sinkTopic;
+    private final Pattern sourcePattern;
+    private final Produced<K, V> produced;
+    private final Consumed<K, V> consumed;
+    private final Materialized<K, V, S> materialized;
+    private final ProcessorSupplier<K, V> processorSupplier;
+    private MaterializedInternal<K, V, S> materializedInternal;
+    private ProducedInternal<K, V> producedInternal;
+    private ConsumedInternal<K, V> consumedInternal;
```

Comment:
This is from trying to capture all possible values in one object; this will be refactored and pared down to only contain required values in more specific graph nodes.